activestorage/test/jobs/transform_job_test.rb

```diff
@@ -27,6 +27,17 @@ class ActiveStorage::TransformJobTest < ActiveJob::TestCase
     end
   end
 
+  test "creates a variant associated with a preview_image for a previewable file" do
+    @blob = create_file_blob(filename: "report.pdf", content_type: "application/pdf")
+    transformations = { resize_to_limit: [100, 100] }
+
+    assert_changes -> { @blob.reload.preview_image.variant(transformations)&.send(:processed?) }, from: nil, to: true do
```

Comment:
since the preview_image does not exist before calling the job, variant(transformations) returns nil, that's why i had to add th &.send